packages/next-auth/src/next/index.ts

```diff
@@ -143,7 +153,7 @@ export async function unstable_getServerSession(
   } else {
     req = args[0]
     res = args[1]
-    options = args[2]
+    options = Object.assign(args[2], { providers: [] })
```

Comment:
Is it intentional that we are modifying `args[2]` here? I recently updated next-auth and started getting some odd behavior where I've got places in my app using my `authOptions` object and it's fine at first, but eventually the providers got cleared out...this is the reason for that.

I suspect we meant to do `options = Object.assign({}, args[2], { providers: [] })` instead?

Reply:
I've opened a [PR](https://github.com/nextauthjs/next-auth/pull/5973) that should fix this.